src/ui/Form/DateField.tsx

```diff
@@ -47,28 +48,32 @@ function DatePicker<FormValues extends FieldValues>({
         // onDayChange won't work with user keyboard's input
         dayPickerProps={{ ...dayPickerProps, onDayClick: field.onChange }}
         placeholder=" "
-        formatDate={formatDate}
-        parseDate={parseDate}
+        formatDate={(date, format) => dayjs(date).format(format)}
+        parseDate={(dateStr, format) => {
+          const parsed = dayjs(dateStr, format);
+          return parsed.isValid() ? parsed.toDate() : undefined;
+        }}
       />
 
       <StyledLabel
         isInvalid={fieldState.invalid}
-        isActive={!isEmptyString(value)}
+        isActive={!isEmptyString(field.value)}
       >
         {label}
       </StyledLabel>
 
       <Form.TextWrapper>
-        {fieldState.invalid && (
-          <Form.ErrorMessage>{fieldState.error?.message}</Form.ErrorMessage>
-        )}
-        {!fieldState.invalid && (
-          <Form.Description>{inputDescription}</Form.Description>
+        {fieldState.invalid ? (
```

Comment:
Same as checkbox, helper text should be visible.

Reply:
Same as [here](https://github.com/Andrewnt219/rent-near-me/pull/96#discussion_r743710763)